pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -97,11 +98,51 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
+	// Explicit combinations in the Matrix
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
+
+	// Iterate over matrix.params and compute count of all generated combinations
+	count := m.countMatrixParamsCombinations()
+
+	// Add any additional combinations generated from matrix include params
+	count += m.countNewCombinationsFromInclude()
+
+	return count
+}
+
+// countMatrixParamsCombinations returns the count of combinations of Parameters generated from the matrix
+// parameters
+func (m *Matrix) countMatrixParamsCombinations() int {
+	if !m.hasParams() {
+		return 0
+	}
+	count := 1
+	for _, param := range m.Params {
+		count *= len(param.Value.ArrayVal)
+	}
+	return count
+}
+
+// countNewCombinationsFromInclude returns the count of combinations of Parameters generated from the matrix
+// include parameters
+func (m *Matrix) countNewCombinationsFromInclude() int {
 	count := 0
-	if m.hasParams() {
-		count = 1
-		for _, param := range m.Params {
-			count *= len(param.Value.ArrayVal)
+	if !m.hasInclude() {
+		return count
+	}
```

Comment:
```suggestion
	if !m.hasInclude() {
		return 0
	}
	if !m.hasParams() {
		return len(m.Include)
	}
	count := 0 
```
(remember to delete "count := 0" in line 131